src/app/getting-started/getting-started.component.ts

```diff
@@ -144,14 +170,44 @@ export class GettingStartedComponent implements OnDestroy, OnInit {
   /**
    * Link GitHub and/or OpenShift accounts
    */
-  connectAccounts(): void {
-    if (this.authGitHub && !this.gitHubLinked && this.authOpenShift && !this.openShiftLinked) {
-      this.providerService.linkAll(window.location.origin + "/_gettingstarted?wait=true");
-    } else if (this.authGitHub && !this.gitHubLinked) {
-      this.providerService.linkGitHub(window.location.origin + "/_gettingstarted?wait=true");
-    } else if (this.authOpenShift && !this.openShiftLinked) {
+  connectAccounts(): boolean {
+
+    // This method is called on form submit.
+    // If Github Or Github+OSO is being linked, then form POST is done to /api/token/link.
+    // If only OSO is being linked, then form POST is prevented and a redirection is used.
+    
+  
+    if ( (this.authOpenShift && !this.openShiftLinked) && (this.authGitHub && !this.gitHubLinked) ){  
+
+        console.log("Both openshift and github to be linked");
+
+        let finalRedirect = window.location.origin + "/_gettingstarted?wait=true";     
+
+        let parsedToken = jwt_decode(this.auth.getToken());
+      
+        // code repition, but temporary.
+        // Will soon migrate this to use the new way of 
+        // linking using the auth service.
+        let url = `${this.oldLinkingServer}/link/session?`
+          + "clientSession=" + parsedToken.client_session
+          + "&sessionState=" + parsedToken.session_state
+          + "&redirect=" + finalRedirect // brings us back to Getting Started.
+          + "&provider=openshift-v3";
```

Comment:
Looks like this code probably belongs in the prvider service, not in the getting started compponent?

Reply:
Thanks @dlabrecq , moved it to `provider service`

Reply:
I see a lot of new code, to support a form submit, but existing code has not been refactored or removed.